x-pack/plugins/ingest_manager/server/services/agents/acks.ts

```diff
@@ -64,10 +80,22 @@ export async function acknowledgeAgentActions(
       return data?.config?.revision > acc ? data?.config?.revision : acc;
     }, agent.config_revision || 0);
 
-    await soClient.update<AgentSOAttributes>(AGENT_SAVED_OBJECT_TYPE, agent.id, {
-      actions: matchedUpdatedActions,
-      config_revision: configRevision,
-    });
+    await soClient.bulkUpdate<AgentSOAttributes | AgentActionSOAttributes>([
+      {
+        type: AGENT_SAVED_OBJECT_TYPE,
+        id: agent.id,
+        attributes: {
+          config_revision: configRevision,
+        },
+      },
+      ...matchedUpdatedActions.map(updatedAction => ({
+        type: AGENT_ACTION_SAVED_OBJECT_TYPE,
+        id: updatedAction.id,
+        attributes: {
+          sent_at: updatedAction.sent_at,
+        },
+      })),
+    ]);
```

Comment:
Something breaking a function up may make you jump around, what do you think? Some folks think it makes it readable, others may think otherwise. It seems like the function is long and do a few things.

Reply:
👍I moved a few things outside this function